go/vt/vttablet/tabletserver/planbuilder/plan.go

```diff
@@ -218,13 +211,22 @@ func Build(statement sqlparser.Statement, tables map[string]*schema.Table, isRes
 		}, nil
 	case *sqlparser.Select:
 		plan, err = analyzeSelect(stmt, tables)
```

Comment:
shouldn't we check the err here before moving on?

Reply:
this is based on assumption that only one of them will be non-nil and not both.
I can do the check if you think otherwise.

Reply:
looks like an error to me like this